tests/test_utils.py

```diff
@@ -174,421 +282,128 @@ def test_parse_log_components(ns_3_compiled_debug, config):
-              '+0.000000000s -1 WifiPhy:SetChannelNumber():  \n'
-              ]
-    with open(data_dir, "w") as f:
-        f.writelines(log_ls)
-
-    parse_list = parse_logs(data_dir)
-
-    expected_list = [
-        {
-            'Time': 0.000000000,
-            'Context': '-1',
-            'Extended_context': 'mac=00:00:00:00:00:00',
-            'Component': 'FrameExchangeManager',
-            'Function': 'SetWifiMac',
-            'Arguments': '0x5576595683e0, 0x557659603820',
-            'Severity_class': 'FUNCTION',
-            'Message': ''
-        },
-        {
-            'Time': 0.045510017,
-            'Context': '1',
-            'Extended_context': 'mac=00:00:00:00:00:01',
-            'Component': 'FrameExchangeManager',
-            'Function': 'RxStartIndication',
-            'Arguments': '0x5576595683e0, "PSDU reception started for ", +76us, " (txVector: ", txpwrlvl: 17 preamble: LONG channel width: 20 GI: 800 NTx: 97 Ness: 0 MPDU aggregation: 0 STBC: 0 FEC coding: BCC mode: OfdmRate6Mbps Nss: 1, ")"',
-            'Severity_class': 'FUNCTION',
-            'Message': ''
-        },
-        {
-            'Time': 0.000000000,
-            'Context': '-1',
-            'Extended_context': None,
-            'Component': 'WifiPhy',
-            'Function': 'WifiPhy',
-            'Arguments': '0x557659571bd0',
-            'Severity_class': 'FUNCTION',
-            'Message': ''
-        },
-        {
-            'Time': 0.000000000,
-            'Context': '-1',
-            'Extended_context': None,
-            'Component': 'WifiPhy',
-            'Function': 'SetChannelNumber',
-            'Arguments': '',
-            'Severity_class': 'DEBUG',
-            'Message': 'Saving channel number configuration for initialization'
-        },
-        {
-            'Time': 0.506390387,
-            'Context': '1',
-            'Extended_context': 'mac=00:00:00:00:00:01',
-            'Component': 'FrameExchangeManager',
-            'Function': 'StartTransmission',
-            'Arguments': '',
-            'Severity_class': 'DEBUG',
-            'Message': 'MPDU payload size=36, to=00:00:00:00:00:02, seq=16'
-        },
-        {
-            'Time': 0.000000000,
-            'Context': '-1',
-            'Extended_context': None,
-            'Component': 'WifiPhy',
-            'Function': 'SetChannelNumber',
-            'Arguments': '',
-            'Severity_class': 'DEBUG',
-            'Message': '()[]./-='
-        }
-    ]
-
-    assert len(parse_list) == 6
-    assert all([actual == expected for actual, expected in zip(parse_list, expected_list)])
-
-    os.remove(data_dir)
+        parse_list = parse_logs(data_dir)
+        assert len(parse_list) == len(expected_list)
+        assert all([actual == expected for actual, expected in zip(parse_list, expected_list)])
+    except RuntimeError:
+        print('Test for parse_logs failed')
```

Comment:
All built-in, non-system-exiting exceptions are derived from `Exception`. All user-defined exceptions should also be derived from this class.
I have one doubt: does the try block make tests succeed by default?
```suggestion
    except Exception as e:
        print('Test for parse_logs failed')
        print(e)
```

Reply:
> I have one doubt: does the try block make tests succeed by default?

This might happen. To avoid this I changed try;except;finally to try;finally. Now, the tests will not succeed by default and the 'finally' segment will always be executed.

Reply:
Fixed in 1d87256.